rdt/transformers/categorical.py

```diff
@@ -455,14 +459,39 @@ def _fit(self, data):
     def _transform(self, data):
         """Replace each category with its corresponding integer value.
 
+        If a category has not been seen before, a random value is assigned.
+
         Args:
             data (pandas.Series):
                 Data to transform.
 
         Returns:
-            numpy.ndarray:
+            pd.Series.
+
+        Raises:
+            `NotFittedError` when no categories have been fitted and `data` is not empty.
         """
-        return pd.Series(data).map(self.categories_to_values)
+        # Find the data categories that were not seen during fitting. Convert all nan's
+        # to np.nan to avoid having nan values with different ids.
+        mapped = data.fillna(np.nan).map(self.categories_to_values)
+        is_null = mapped.isna()
+        if is_null.any():
+            if len(self.categories_to_values) == 0:
+                raise NotFittedError('No categories have been fitted.')
```

Comment:
Can you move this to the first thing? I think we want to raise the error whether or not there are nulls right?

Reply:
As mentioned above, I'm deleting the check.